### PR TITLE
feat: add anr xapp

### PR DESCRIPTION
**Pull request by colinmhoward**

### CODE REVIEW

File: charts/xapp-anr/.ct-install-ignore
Comment (Wielewout) [inline charts/xapp-anr/.ct-install-ignore:1]: For now this needs to be there, but we should look into a way to make this testable in the longer term (ie separately installing kafka in the flow; maybe a separate flow for xapps even?)
To be disccussed outside the scope of this PR.

File: charts/xapp-anr/Chart.yaml
Comment (Wielewout) [inline charts/xapp-anr/Chart.yaml:1]: Remove the comments, they don't add any value after the chart has been created...

File: charts/xapp-anr/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+type: application
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
```
Comment (Wielewout) [inline charts/xapp-anr/Chart.yaml]: Add a comment for renovate so we get automatic PRs with updates.

```suggestion
# renovate: image=accelleran/xapp-anr
appVersion: "0.1.0"
```

File: charts/xapp-anr/values.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  name: ""
+  kafka:
+    enabled: true
+    hostname: ""
+    port: 0
+
+image:
+  repository: accelleran/xapp-anr
+  pullPolicy: IfNotPresent
+  tag: "0.1.0"
```
Comment (Wielewout) [inline charts/xapp-anr/values.yaml]: If we leave this out with an empty string, then the app version is used as the default value.

```suggestion
  tag: ""
```

File: charts/xapp-anr/values.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+imagePullSecrets:
+  - name: accelleran-secret
+
+accelleranLicense:
+  enabled: false
```
Comment (Wielewout) [inline charts/xapp-anr/values.yaml]: Enable the license by default.

File: charts/xapp-anr/templates/rbac.yaml
Comment (Wielewout) [inline charts/xapp-anr/templates/rbac.yaml:1]: I would strip out everything related to rbac as we don't need it (it is already disabled by default).

File: charts/xapp-anr/values.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+fullnameOverride: ""
+
+replicaCount: 1
+
+bootstrap:
+  create: true
+  name: ""
+  kafka:
+    enabled: true
+    hostname: ""
```
Comment (Wielewout) [inline charts/xapp-anr/values.yaml:13]: It's maybe worth looking into #142 first so we would get an error message when this field is not provided.

**Review by Wielewout — COMMENTED** [review]
A couple of small remarks. Looks already quite good in general.

Small note: once this is merged that's also the first release of this chart. So we maybe should be looking into an update in the common charts with #142 first before releasing this chart.

File: charts/xapp-anr/.ct-install-ignore
Comment (ben-foxmoore) [reply charts/xapp-anr/.ct-install-ignore:1]: Yes, we'll need to find a way to pre-provision certain components before testing individual charts - essentially it's the same need as the CU with NATS/Redis, but I don't want us to have to package Kafka into each xApp Helm chart. Maybe worth a discussion 'issue' here in the repo?

File: charts/xapp-anr/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+################################################################################
+
+Accelleran ANR Xapp {{ $.Chart.Version }}
+
+Deployed ANR XApp [{{ $.Release.Name }}]
```
Comment (ben-foxmoore) [inline charts/xapp-anr/templates/NOTES.txt]: `xApp` rather than `XApp` 😉

File: charts/xapp-anr/Chart.yaml
Comment (colinmhoward) [reply charts/xapp-anr/Chart.yaml:1]: OK - done

File: charts/xapp-anr/Chart.yaml (hunk 1 above)
Comment (colinmhoward) [reply charts/xapp-anr/Chart.yaml]: Ah yes - done

File: charts/xapp-anr/values.yaml (hunk 2 above)
Comment (colinmhoward) [reply charts/xapp-anr/values.yaml]: Yes OK

File: charts/xapp-anr/values.yaml (hunk 3 above)
Comment (colinmhoward) [reply charts/xapp-anr/values.yaml]: OK - done

File: charts/xapp-anr/templates/rbac.yaml
Comment (colinmhoward) [reply charts/xapp-anr/templates/rbac.yaml:1]: OK - done

File: charts/xapp-anr/values.yaml (hunk 4 above)
Comment (colinmhoward) [reply charts/xapp-anr/values.yaml:13]: OK - do I have to change anything in this part of the values file?

Comment (Wielewout) [reply charts/xapp-anr/values.yaml:13]: No, it's already good

File: charts/xapp-anr/.ct-install-ignore
Comment (Wielewout) [reply charts/xapp-anr/.ct-install-ignore:1]: Created issue #162

**Comment by ben-foxmoore** [comment]
@colinmhoward I think we're missing a commit with your latest changes?

**Review by ben-foxmoore — DISMISSED** [review]
@colinmhoward just marking this as "changes requested" so that the status shows that this has been reviewed, rather than awaiting review.

File: charts/xapp-anr/templates/NOTES.txt (hunk 5 above)
Comment (colinmhoward) [reply charts/xapp-anr/templates/NOTES.txt]: OK :)

**Review by Wielewout — APPROVED** [review]

**Review by ben-foxmoore — COMMENTED** [review]
Afraid I won't have time to take a proper look at this 😞

**Pull request merged**